src/expressions.rs

```diff
@@ -36,7 +35,17 @@ fn find_closest_state(inputs: &[Series]) -> PolarsResult<Series> {
             for (lhs_opt_val, rhs_opt_val) in lhs_arr.iter().zip(rhs_arr.iter()) {
                 match (lhs_opt_val, rhs_opt_val) {
                     (Some(lhs_val), Some(rhs_val)) => {
-                        let res = &geocoder.search((*lhs_val, *rhs_val)).record.admin1;
+                        let record = &geocoder.search((*lhs_val, *rhs_val)).record;
+                        let res: &String;
+                        if return_type == "city" {
+                            res = &record.name;
+                        } else if return_type == "state" {
+                            res = &record.admin1;
+                        } else if return_type == "country" {
+                            res = &record.cc;
+                        } else {
+                            res = &record.name;
+                        }
```

Comment:
this seems error prone

Instead of taking a string argument, could this function take an Enum?

Reply:
Good idea, let me fix it

Reply:
Please note that I removed 'reverse_geocode' function, so now 'reverse_geocode_chunks' handles cities 'find_closest_city' as well. If there was any specific reason to use different implementation for different record types, please let me know